heuristic/repair_operators.py

```diff
@@ -69,16 +72,19 @@ def worst_week_repair(shifts_in_week, competencies, t_covered_by_shift, employee
         ]
         competencies_needed = tuple(set(y_s))
 
-        deviation_contracted_hours = {
-            e: (
-                sum(state.soft_vars["deviation_contracted_hours"][e, j] for j in weeks)
-                - 0 * (competency_level - len(competencies_needed))
-            )
-            for (competency_level, e) in employee_with_competency_combination[competencies_needed]
-            if (sum(state.x[e, t, v] for t, v in shifts_at_day[int(shift[0] / 24)])) == 0
-        }
+        
+        deviation_contracted_hours = {e: (sum(state.soft_vars["deviation_contracted_hours"][e,j] for j in weeks)  
+                                    - (competency_level - len(competencies_needed)))
+                                    + (20 if (sum(state.soft_vars["deviation_contracted_hours"][e,j] for j in weeks) - shift[1] >= 8.5) else 0)
+                                    + (20 if (sum(state.soft_vars["deviation_contracted_hours"][e,j] for j in weeks) - shift[1] <= 17) else 0)
+                                    + (100 if (sum(state.soft_vars["deviation_contracted_hours"][e,j] for j in weeks) - shift[1]  == 0) else 0)
```

Comment:
What is the intention here? To reward: 
- +20 if the deviation is longer than 17
- +20 if the deviation is shorter than 8.5
- +40 if the deviation is between 8.5 and 17
- +100 if the deviation is 0
? 

Why exactly this reward scheme?

Reply:
It is mostly a test. Did improve the solution, but should be improved upon.

Reply:
I see! There is definitely potential for improvement here, both with regards to the actual weighting and performance (4 loops for calculating the sum seems 2 more than needed). If you feel confident enough about this we can merge it now, but then we need to come back to this later. Can you add a Trello-card for this?

Reply:
Done